File: app/src/main/java/fup/prototype/robprototype/di/SearchModule.java
package fup.prototype.robprototype.di;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import fup.prototype.robprototype.search.MainActivity;
import fup.prototype.robprototype.search.MainFragment;

@Module
public abstract class SearchModule {

    @ContributesAndroidInjector
    abstract MainActivity bindMainActivity();

    @ContributesAndroidInjector
    abstract MainFragment bindNewMainFragment();
}

File: app/src/main/java/fup/prototype/robprototype/di/SearchModule.kt
package fup.prototype.robprototype.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import fup.prototype.robprototype.search.SearchActivity
import fup.prototype.robprototype.search.SearchFragment

@Module
abstract class SearchModule {

    @ContributesAndroidInjector
    abstract fun bindSearchActivity(): SearchActivity

    @ContributesAndroidInjector
    abstract fun bindKtSearchFragment(): SearchFragment

}